src/core/src/op/grid_sample.cpp

```diff
@@ -4,27 +4,75 @@
 
 #include "openvino/op/grid_sample.hpp"
 
+#include "element_visitor.hpp"
 #include "grid_sample_shape_inference.hpp"
 #include "itt.hpp"
-#include "ngraph/validation_util.hpp"
 #include "openvino/reference/grid_sample.hpp"
+#include "validation_util.hpp"
 
 namespace ov {
-op::v9::GridSample::GridSample(const Output<Node>& data, const Output<Node>& grid, const Attributes& attributes)
+namespace op {
+namespace v9 {
+
+struct Evaluate : element::NoAction<bool> {
+    using element::NoAction<bool>::visit;
+
+    template <element::Type_t ET, class T = fundamental_type_for<ET>>
+    static result_type visit(Tensor& output,
+                             const Tensor& data,
+                             const Tensor& grid,
+                             const Shape& data_shape,
+                             const Shape& grid_shape,
+                             const GridSample::Attributes& attributes) {
+        using namespace ov::element;
+        return IfTypeOf<f32>::apply<EvalByGridType>(grid.get_element_type(),
+                                                    output.data<T>(),
+                                                    data.data<const T>(),
+                                                    grid,
+                                                    data_shape,
+                                                    grid_shape,
+                                                    attributes);
+    }
+
+private:
+    struct EvalByGridType : public element::NoAction<bool> {
```

Comment:
What the purpose was to make struct private? In other tests same looked like namespace::Evaluate (e.g. greater::Evaluate)

Reply:
These part is not used from class member function, is like 2nd stage of evaluate and made it intentionally private to not be visible. As this is internal helper it can be change if required but can be hidden now.